app/src/main/kotlin/com/tailoredapps/countriesexample/detail/recyclerview/DetailViewHolder.kt

```diff
@@ -18,10 +18,11 @@ import androidx.recyclerview.widget.RecyclerView
 import com.tailoredapps.countriesexample.databinding.ItemDetailBinding
 
 class DetailViewHolder(
-    private val binding: ItemDetailBinding
+    private val binding: ItemDetailBinding,
+    private val interaction: ((DetailAdapterInteraction) -> Unit)
 ) : RecyclerView.ViewHolder(binding.root) {
 
-    fun bind(item: DetailAdapterItem, interaction: ((DetailAdapterInteraction) -> Unit)) {
+    fun bind(item: DetailAdapterItem) {
```

Comment:
What's the idea behind changing the onClick listener lambda (`interaction`) to a class field?
Isn't that one moew reference in the whole system?

I would have seen the place to _bind_ the lambda to the onClickListener to be the bind method.

Another (very weak) argument I would have against moving this to be a class field, is, that when reusing this viewHolder in a case where no onClickItemListener is set, it still references the lambda, whereas when this would be handled to be a parameter to the bind method, the onClickListener could be set to null and the reference from this ViewHolder would be gone.

Reply:
I actually cannot remember anymore, PR is too stale 😛 . I think I actually wanted to only set the click listener on init, so that it is not reset on every viewholder bind... but I guess I forgot that 😅